typhos/utils.py

```diff
@@ -1048,3 +1050,65 @@ def get_text(item):
 def nullcontext():
     """Stand-in for py3.7's contextlib.nullcontext"""
     yield
+
+
+def get_component(obj):
+    """
+    Get the component that made the given object.
+
+    Parameters
+    ----------
+    obj : ophyd.OphydItem
+        The ophyd item for which to get the component.
+
+    Returns
+    -------
+    component : ophyd.Component
+        The component, if available.
+    """
+    if obj.parent is None:
+        return None
+
+    return getattr(type(obj.parent), obj.attr_name, None)
+
+
+def get_variety_metadata(cpt):
```

Comment:
I see there are some duplicated functions here. I assume this is to avoid weird dependency chains?

Reply:
Yes - we don't want typhos to depend on pcdsdevices. The only common place, otherwise, would be to move variety metadata stuff to either pcdsutils (or less likely, ophyd itself).